### PR TITLE
Add eddsa support

### PR DESCRIPTION
**Pull request by ju-cu**
Add support for validating EdDSA signatures in tokens

### CODE REVIEW

File: src/main/java/io/curity/oauth/AbstractJwtValidator.java [hunk 1]
```diff
@@ -191,7 +205,13 @@ private boolean verifySignature(byte[] signingInput, byte[] signature, PublicKey
 
     private boolean canRecognizeAlg(String alg)
     {
-        return alg.equals("RS256");
+        switch (alg) {
```
Comment (travisspencer) [inline src/main/java/io/curity/oauth/AbstractJwtValidator.java]: switch expressions look better IMO but that's a preference thing

File: src/main/java/io/curity/oauth/AbstractJwtValidator.java [hunk 2]
```diff
@@ -172,11 +173,24 @@ private byte[] convertToBytes(String input)
         return bytes;
     }
 
-    private boolean verifySignature(byte[] signingInput, byte[] signature, PublicKey publicKey)
+    private boolean verifySignature(String algorithm, byte[] signingInput, byte[] signature, PublicKey publicKey)
     {
         try
         {
-            Signature verifier = Signature.getInstance("SHA256withRSA");
+            Signature verifier;
```
Comment (travisspencer) [inline src/main/java/io/curity/oauth/AbstractJwtValidator.java]: you can assign this the value of a switch expression if you want

File: src/main/java/io/curity/oauth/EdDSAPublicKeyCreator.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import java.security.spec.EdECPublicKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.NamedParameterSpec;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+
+public class EdDSAPublicKeyCreator {
```
Comment (travisspencer) [inline src/main/java/io/curity/oauth/EdDSAPublicKeyCreator.java]: final

File: src/main/java/io/curity/oauth/EdDSAPublicKeyCreator.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    static PublicKey createPublicKey(String curveName, String publicKeyJwt) throws InvalidKeySpecException, NoSuchAlgorithmException
+    {
+        Base64.Decoder decoder = Base64.getUrlDecoder();
+        byte[] publicKeyBytes = decoder.decode(publicKeyJwt);
+        int b = publicKeyBytes.length;
+
+        if (b != 32 && b != 57) {
```
Comment (travisspencer) [inline src/main/java/io/curity/oauth/EdDSAPublicKeyCreator.java:21]: formatting

**Review by mtrojanowski — APPROVED** [review]

**Pull request merged**